merlin/dag/node.py

```diff
@@ -613,7 +616,7 @@ def iter_nodes(nodes):
 
 
 # output node (bottom) -> selection leaf nodes (top)
-def preorder_iter_nodes(nodes):
+def preorder_iter_nodes(nodes, flatten_subgraphs=False):
```

Comment:
Noticing that this boolean param is added to multiple functions makes me wonder if there's a missing concept here. Back in the day (before `Graph`, `Node`, etc existed and everything was just a `ColumnGroup`), it probably made more sense to have a plain function instead of a method, but now this is starting to look like either:
* we're iterating through two different kinds of things that require different iteration behavior and could use polymorphism to achieve that
* we're doing two different kinds of iteration over the same kind of thing and we could use two methods to capture those behaviors